docker/test/stress/run.sh

```diff
@@ -101,7 +101,12 @@ EOL
 
 function stop()
 {
-    clickhouse stop
+    clickhouse stop --do-not-kill && return
+    # We failed to stop the server with SIGTERM. Maybe it hang, let's collect stacktraces.
+    kill -TERM "$(pidof gdb)" ||:
+    sleep 5
+    gdb -batch -ex 'thread apply all backtrace' -p "$(cat /var/run/clickhouse-server/clickhouse-server.pid)" ||:
```

Comment:
Maybe it worth redirect to `gdb.log` too?

Reply:
Hm, yes, it would be better to redirect it and avoid printing tons of stacktarces to `runlog.log` (but I just don't care, that's enough for me that we write these stacktraces _somewhere_)

Reply:
https://github.com/ClickHouse/ClickHouse/pull/38749/commits/8e9c1eaa73b78390d78a25b2867bcfc0d55412a7